### PR TITLE
Add MISR driver

### PR DESCRIPTION
**Pull request by wavious-jbasista**
Fixes
  - None

Features
  - Add MISR driver
  - Update dfich header from WDDR101_20210907_HW tag

### CODE REVIEW

File: drivers/misr/driver.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <misr/driver.h>
+#include <wddr/memory_map.h>
+#include <kernel/io.h>
+
+#define MISR_READ_START__ADDR   DDR_DFICH_DQ0_W3_TO_W0_MISR_LO_STA__ADR
+#define MISR_READ_OFFSET        DDR_DFICH_DQ0_W7_TO_W4_MISR_LO_STA__ADR - \
```
Comment (wavious-wpatty) [inline drivers/misr/driver.c:12]: Always wrap "#define" in parentheses.

File: drivers/misr/driver.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFICH_MISR_0_CFG_LFSR_LD_SEED, 0x1);
+    reg_write(WDDR_MEMORY_MAP_DFI_CH0 + DDR_DFICH_MISR_0_CFG__ADR, reg_val);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_DFICH_MISR_0_CFG_LFSR_LD_SEED, 0x0);
+    reg_write(WDDR_MEMORY_MAP_DFI_CH0 + DDR_DFICH_MISR_0_CFG__ADR, reg_val);
+}
+
+void misr_read_reg_if(misr_t misr, uint32_t *low, uint8_t *high)
+{
+    uint32_t lo_reg_addr = WDDR_MEMORY_MAP_DFI_CH0 + MISR_READ_START__ADDR +
+                           MISR_READ_OFFSET * misr;
```
Comment (wavious-wpatty) [inline drivers/misr/driver.c:66]: Perfect example here of why you need parentheses.

File: include/drivers/misr/driver.h [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    MISR_EN_CA  = 0x1,
+    MISR_EN_DQ0 = 0x2,
+    MISR_EN_DQ1 = 0x4,
+    MISR_EN_DQ2 = 0x8,
+    MISR_EN_DQ3 = 0x10,
+    MISR_EN_ALL = 0x1F
+} misr_en_t;
+
+/**
+ * @brief   MISR Enumeration
```
Comment (wavious-wpatty) [inline include/drivers/misr/driver.h:33]: Make this a little more descriptive in the description. I would say how these map to different HW MISRs or something. Just more clear about what misr_t is supposed to enumerate.

File: include/drivers/misr/driver.h [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+ * @brief   MISR Load LFSR Seed Register Interface
+ *
+ * @details Loads LFSR Seed value at the register interface level.
+ *
+ * @param[in]   low     Lower 32 bits of seed value.
+ * @param[in]   high    Upper 4 bits of seed value.
+ *
+ * @return      void
+ */
+void misr_load_lfsr_seed_reg_if(uint32_t low, uint8_t high);
```
Comment (wavious-wpatty) [inline include/drivers/misr/driver.h:95]: I would just make both high and low 32-bits. HW probably won't change but I think 32-bits for high gives us some extra room to expand in the future if needed. There's no cost of having uint32_t versus uint8_t.

File: include/drivers/misr/driver.h [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+ *
+ * @details Reads current MISR value from the register interface level.
+ *
+ * @param[in]   misr    MISR to read from.
+ * @param[out]  low     Lower 32 bits of seed value.
+ * @param[out]  high    Upper 4 bits of seed value.
+ *
+ * @return      void
+ */
+void misr_read_reg_if(misr_t misr, uint32_t *low, uint8_t *high);
```
Comment (wavious-wpatty) [inline include/drivers/misr/driver.h:108]: same here as comment above.

**Review by wavious-wpatty — CHANGES_REQUESTED** [review]
Looks good. A few small things to update. Nothing major.

File: drivers/misr/driver.c (hunk 1 above)
Comment (wavious-jbasista) [reply drivers/misr/driver.c:12]: Fixed.

File: drivers/misr/driver.c (hunk 2 above)
Comment (wavious-jbasista) [reply drivers/misr/driver.c:66]: This makes sense why my reg_addr was incorrect, will be sure to wrap the defines in parentheses from now on.

File: include/drivers/misr/driver.h (hunk 4 above)
Comment (wavious-jbasista) [reply include/drivers/misr/driver.h:95]: I am ok with this, doesn't hurt to make it uint32_t.  Fixed.

**Pull request closed without merging**